spacetelescope/asdf-standard
# Additional schema tests and resulting cleanup

This is the main course of my cleanup effort.  New tests include:

For each schema:
- Confirm that filename follows our convention thus far (name with lowercase letters, underscores, and numbers, followed by a 3-part version and .yaml extension)
- Confirm presence of YAML header and footer
- Confirm that `$schema` key exists and is set to yaml-schema/draft-01
- Confirm that the `id` key exists, matches the filename, and is unique among all the schemas
- If the `tag` key is present, confirm that it matches the filename, and is unique among all the schemas
- Confirm that the `title` key is present and has a value
- Confirm that the `description` key is present and has a value
- Confirm that examples reference the version of the schema described by the file (and not some earlier version, from a copy/paste)
- Confirm that refs in the description reference the version of the schema described by the file

For each transform schema:
- Confirm that schema includes another transform schema or base transform schema

For the latest version of each schema:
- Confirm that `$ref` values only reference other latest-version schemas
- Confirm that examples only include other latest-version schemas
- Confirm that refs in the description reference only other latest-version schemas

For each metaschema:
- Confirm presence of YAML header and footer
- Confirm valid YAML

For each version map:
- Confirm that filename follows our convention (version_map-x.y.z.yaml)
- Confirm presence of YAML header and footer
- Confirm that the only keys are FILE_FORMAT, YAML_VERSION, and tags
- Confirm that FILE_FORMAT and YAML_VERSION have recognized values (currently only 1.0.0/1.1)
- Confirm that all the tags listed exist
- Confirm that the tag list doesn't contain duplicate names
- Confirm that the tag list is sorted alphabetically by tag name
- Confirm that tags did not disappear between successive version maps (except those that were explicitly deprecated, like transform/domain)

For the latest version map:
- Confirm that tags include the latest version of every non-deprecated tag

These tests revealed many issues that are all resolved here.
